crates/server/src/server/server_future.rs

```diff
@@ -489,7 +489,7 @@ impl<T: RequestHandler> ServerFuture<T> {
         use trust_dns_rustls::tls_server;
 
         let spawner = Handle::current();
-        let dns_hostname = Arc::new(dns_hostname);
+        let dns_hostname: Arc<str> = Arc::from(dns_hostname);
```

Comment:
And same here... although I guess it's actually needed?

Reply:
This is just a problem of how late the actual type is bound, so the compiler was confused.